corehq/apps/locations/tests/test_bulk_management.py

```diff
@@ -0,0 +1,110 @@
+from django.test import TestCase
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.locations.bulk_management import bulk_update_organization
+
+# These example types and trees mirror the information available in the upload files
+
+# TODO What fields should be editable in the upload?
+# Look at what's editable in the UI, especially with commtrack enabled versus disabled
+
+FLAT_LOCATION_TYPES = [
+    # ('name', 'code', 'parent_code', 'shares_cases', 'view_descendants'),
+    ('State', 'state', '', False, False),
+    ('County', 'county', 'state', False, True),
+    ('City', 'city', 'county', True, False),
+]
+BASIC_LOCATION_TREE = [
+    # ('name', 'site_code', 'parent_code', 'location_id', 'external_id', 'latitude', 'longitude'),
+    ('Massachusetts', 'mass', '', '1234', '', '', ''),
+    ('Suffolk', 'suffolk', 'mass', '2345', '', '', ''),
+    ('Boston', 'boston', 'suffolk', '2346', '', '', ''),
+    ('Middlesex', 'middlesex', 'mass', '3456', '', '', ''),
+    ('Cambridge', 'cambridge', 'middlesex', '3457', '', '', ''),
+    ('Florida', 'florida', '', '5432', '', '', ''),
+    ('Duval', 'duval', 'florida', '5433', '', '', ''),
+    ('Jacksonville', 'jacksonville', 'duval', '5434', '', '', ''),
+]
+
+MOVE_SUFFOLK_TO_FLORIDA = [
+    # ('name', 'site_code', 'parent_code', 'location_id', 'external_id', 'latitude', 'longitude'),
+    ('Massachusetts', 'mass', '', '1234', '', '', ''),
+    # this is the only changed line:
+    ('Suffolk', 'suffolk', 'florida', '2345', '', '', ''),
+    ('Boston', 'boston', 'suffolk', '2346', '', '', ''),
+    ('Middlesex', 'middlesex', 'mass', '3456', '', '', ''),
+    ('Cambridge', 'cambridge', 'middlesex', '3457', '', '', ''),
+    ('Florida', 'florida', '', '5432', '', '', ''),
+    ('Duval', 'duval', 'florida', '5433', '', '', ''),
+    ('Jacksonville', 'jacksonville', 'duval', '5434', '', '', ''),
+]
+
+DELETE_SUFFOLK = [
+    # ('name', 'site_code', 'parent_code', 'location_id', 'external_id', 'latitude', 'longitude', 'deleted'),
+    ('Massachusetts', 'mass', '', '1234', '', '', ''),
+    # this is the only changed line:
+    ('Suffolk', 'suffolk', 'florida', '2345', '', '', '', 'yes'),
+    ('Boston', 'boston', 'suffolk', '2346', '', '', '', 'yes'),
+    ('Middlesex', 'middlesex', 'mass', '3456', '', '', ''),
+    ('Cambridge', 'cambridge', 'middlesex', '3457', '', '', ''),
+    ('Florida', 'florida', '', '5432', '', '', ''),
+    ('Duval', 'duval', 'florida', '5433', '', '', ''),
+    ('Jacksonville', 'jacksonville', 'duval', '5434', '', '', ''),
+]
```

Comment:
@sravfeyn you asked for examples of uploads, and I figure if we're doing that, we might as well just turn those into tests.  This is a basic format that I intend to exactly mirror the contents of the eventual excel uploads (one sheet for location types, one for locations).

The idea with these tests is that you have one pair of data structures representing the initial state of the organization (location types plus locations), and a second pair of data structures reflecting the changes the user would like applied.  The tests can then perform the set up to get the initial location state, call the bulk upload function with the desired end state, then assert that the final state matches that second pair of data structures.

This will probably get more interesting with validation, but I think most of that can happen without hitting the database, so that should go in a `SimpleTestCase`.